pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -260,6 +313,12 @@ func (nc *NodeController) SyncCloudNodes() error {
 			glog.Errorf("Delete node %s error: %v", nodeID, err)
 		}
 		nc.deletePods(nodeID)
+		nodesChanged = true
+	}
+
+	// Make external services aware of nodes currently present in the cluster.
+	if nodesChanged {
```

Comment:
why is this part of the node controller? It seems like the service scan and reconciliation is completely distinct from the node controller synchronization.